contribute/design/1.x/cstor-operator/doc.md

```diff
@@ -544,19 +542,164 @@ spec:
+      overProvisioning: false
+      compression: false
+```
+
+### Disk Replacement
+
+* “As an OpenEBS user, I should be able to replace existing block devices on CSPC to perform disk replacement operations.”
+
+Steps To Be Performed By User:
+
+1. `kubectl edit cspc your_cspc_name`
+
+2. Update the existing block devices with new block devices in the CSPC spec against the correct Kubernetes nodes.
+
+
+**Consider following example for mirror pool disk replacement**
+
+
+Current CSPC is following:
+This CSPC corresponds to a mirror pool on node `kubernetes-node1`.
+
+```yaml
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirror
+spec:
+  pools:
+  - nodeSelector:
+      kubernetes.io/hostname: kubernetes-node1
+    raidGroups:
+    - type: mirror
+      isWriteCache: false
+      isSpare: false
+      isReadCache: false
+      blockDevices:
+
+      - blockDeviceName: node1-bd-1
+
+      - blockDeviceName: node1-bd-2
+
+    poolConfig:
+      cacheFile: /tmp/pool1.cache
+      defaultRaidGroupType: mirror
+      overProvisioning: false
+      compression: false
+```
+
+Replacing block devices in mirror pool -- the spec will look
+following:
+
+```yaml
+apiVersion: openebs.io/v1alpha1
+kind: CStorPoolCluster
+metadata:
+  name: cstor-pool-mirror
+spec:
+  pools:
+  - nodeSelector:
+      kubernetes.io/hostname: kubernetes-node1
+    raidGroups:
+    - type: mirror
+      isWriteCache: false
+      isSpare: false
+      isReadCache: false
+      blockDevices:
+
+      - blockDeviceName: node1-bd-3
+
+      - blockDeviceName: node1-bd-2
+
+    poolConfig:
+      cacheFile: /tmp/pool1.cache
       defaultRaidGroupType: mirror
       overProvisioning: false
       compression: false
 ```
+In the above CSPC spec node1-bd-1 is replaced with node1-bd-3
+
+Disk Replacement Validations:
+
+Once user modifies the CSPC spec to trigger disk replacement process below are
+the validations done by the admission-server to restrict invalid changes made to
+CSPC
+
+Below are validations on CSPC by admission server
+
+1. Not more than one block device should be replaced simultaneously in the same
+   raid group.
+
+2. Replacing block devices should not be already in use by the same cStor pools.
+
+3. Replacing another block device in the raid group is not allowed when any
+   of the block device in the same raid group is undergoing replacement.
+
+4. Steps to validate whether someone(other CSPC or local PV) already claimed
+   replacing block device
+
+   4.1 Verify is there any claim create for replacing block device. If there are
+       no claims for replacing block device jump to step 5.
+
+   4.2 If replacing block device was already claimed and if that claim doesn't
+       have CSPC label (or) different CSPC label(i.e block device belongs to
+       some other CSPC) then reject replacement request.
+
+   4.3 If replacing block device has self CSPC label then check is that replacing
```

Comment:
mention as `no BD in this CSPC should have BDC with annotation openebs.io/bd-predecessor as replacing BD`